pkg/didcomm/protocol/introduce/service.go

```diff
@@ -68,28 +69,54 @@ type Service struct {
 	service.Message
 	store     storage.Store
 	callbacks chan callback
+	ctx       internalContext
+	stop      chan struct{}
+	mu        sync.Mutex
```

Comment:
i would name the mutex similar to the var it is protecting.

Reply:
changed to `closedMutex`